tests/system/Validation/ValidationTest.php

```diff
@@ -343,6 +343,101 @@ public function testClosureRuleWithLabel(): void
         );
     }
 
+    /**
+     * Validation rule1
+     *
+     * @param mixed $value
+     */
+    public function rule1($value)
+    {
+        return $value === 'abc';
+    }
+
+    public function testCallableRule(): void
+    {
+        $this->validation->setRules(
+            [
+                'foo' => ['required', [$this, 'rule1']],
+            ],
+            [
+                // Errors
+                'foo' => [
+                    // Specify the array key for the callable rule.
+                    1 => 'The value is not "abc"',
+                ],
+            ],
+        );
+
+        $data   = ['foo' => 'xyz'];
+        $result = $this->validation->run($data);
+
+        $this->assertFalse($result);
+        $this->assertSame(
+            ['foo' => 'The value is not "abc"'],
+            $this->validation->getErrors()
+        );
+        $this->assertSame([], $this->validation->getValidated());
+    }
+
+    /**
+     * Validation rule1
+     *
+     * @param mixed $value
+     */
+    public function rule2($value, array $data, ?string &$error, string $field)
+    {
+        if ($value !== 'abc') {
+            $error = 'The ' . $field . ' value is not "abc"';
+
+            return false;
+        }
+
+        return true;
+    }
+
+    public function testCallableRuleWithParamError(): void
+    {
+        $this->validation->setRules([
+            'foo' => [
+                'required',
+                [$this, 'rule2'],
+            ],
+        ]);
+
+        $data   = ['foo' => 'xyz'];
+        $result = $this->validation->run($data);
+
+        $this->assertFalse($result);
+        $this->assertSame(
+            ['foo' => 'The foo value is not "abc"'],
+            $this->validation->getErrors()
```

Comment:
I thought CS Fixer was enforcing newline parameter trailing commas?
@paulbalandan 
```suggestion
            $this->validation->getErrors(),
```

Reply:
@MGatner It needs PHP 8.0 !
https://wiki.php.net/rfc/trailing_comma_in_parameter_list

Reply:
No. But there's a rule for that but disabled.

Reply:
Oh wow. I've been working in 8 so much I totally forgot this was new 😅 Thanks team.